src/utils/index.ts

```diff
@@ -0,0 +1,156 @@
+import type { InterpolateOptions, Path, PlainObject, Predicate, VeggiesError } from '../types.js'
+
+export const isNumber = (n: unknown): n is number => Number.isFinite(n)
+
+export const isNullsy = (val: unknown): val is undefined | null => val === undefined || val === null
+
+export const isNotNullsy = <T>(val: unknown): val is T => val !== undefined && val !== null
+
+export const isEmpty = (val: unknown): boolean => {
+    if (val === null || val === undefined) return true
+    if (typeof val === 'string' || Array.isArray(val)) return val.length === 0
+    if (typeof val === 'object') return Object.keys(val).length === 0
+    return false
+}
+
+export const isDefined = <T>(toTest: T | undefined | null): toTest is T => {
+    return !!toTest
```

Comment:
Technicaly it would be more explicit to call this function "isTruthy", since it will retun false for a very wide set of defined values. There's already isNotNullish for strict checks on undefined and null so an isDefined function should only check if it's undefined.